packages/sycamore-web/src/lib.rs

```diff
@@ -35,6 +35,28 @@ pub trait Html: GenericNode<EventType = Event, PropertyType = JsValue> {
     const IS_BROWSER: bool;
 }
 
+/// Create a generic `Html` node from a `web_sys::Node`.
+///
+/// # Panics
+/// When G is not either a `DomNode` or a `HydrateNode`.
+pub fn from_web_sys<G: Html>(node: web_sys::Node) -> G {
+    use std::any::{Any, TypeId};
+
+    let type_id = TypeId::of::<G>();
+
+    if TypeId::of::<DomNode>() == type_id {
+        let node = DomNode::from_web_sys(node);
+        return (&node as &dyn Any).downcast_ref().cloned().unwrap();
+    }
+    #[cfg(feature = "hydrate")]
+    if TypeId::of::<HydrateNode>() == type_id {
+        let node = HydrateNode::from_web_sys(node);
+        return (&node as &dyn Any).downcast_ref().cloned().unwrap();
+    }
+
+    panic!("Expected GenericNode to either be a DomNode or a HydrateNode");
```

Comment:
nit:
```suggestion
    panic!("expected GenericNode to either be a DomNode or a HydrateNode");
```
(for consistency)